pkg/e2e/autoscaler/autoscaler.go

```diff
@@ -286,13 +314,41 @@ var _ = g.Describe("[Feature:Machines] Autoscaler should", func() {
 				nodeCounter++
 			}
 
-			glog.Infof("Expecting at least one new node to come up. Initial number of node group nodes: %d. Current number of nodes in the group: %d", nodeGroupInitialTotalNodes, nodeCounter)
-			return nodeCounter > nodeGroupInitialTotalNodes, nil
+			glog.Infof("Expecting %v nodes. Initial number of node group nodes: %d. Current number of nodes in the group: %d", initialNumberOfReplicas0+2, nodeGroupInitialTotalNodes1, nodeCounter)
+			return nodeCounter == int(initialNumberOfReplicas0)+2, nil
 		})
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		g.By("Labeling all nodes belonging to the machineset (after scale up phase)")
-		err = labelMachineSetNodes(client, &targetMachineSet, nodeTestLabel)
+		g.By("Labeling all nodes belonging to the machineset1 (after scale up phase)")
+		err = labelMachineSetNodes(client, &targetMachineSet1, nodeTestLabel1)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("Waiting for cluster to scale up nodes for targetMachineSet0")
+		err = wait.PollImmediate(5*time.Second, e2e.WaitLong, func() (bool, error) {
```

Comment:
This is duplicate code that can be encapsulated with:

```go
waitForNodeScaleUp := func(machineSet *MachineSet, ...) {
}

waitForNodeScaleUp(targetMachineSet0, ...)
waitForNodeScaleUp(targetMachineSet1, ...)
```

And this same pattern can be used for all the places we need to distinguish between the machinesets under test.